client/pages/org-edit-modal/modal.service.js

```diff
@@ -1,36 +1,52 @@
-angular.module('probedock.orgEditModal').factory('orgEditModal', function($modal) {
+angular.module('probedock.orgEditModal').factory('orgEditModal', function(states, $uibModal) {
   return {
-    open: function($scope) {
+    open: function($scope, options) {
+      options = _.extend({}, options);
 
-      var modal = $modal.open({
+      var scope = $scope.$new();
+      _.extend(scope, _.pick(options, 'organization', 'organizationId', 'organizationName'));
+
+      var modal = $uibModal.open({
         templateUrl: '/templates/pages/org-edit-modal/modal.template.html',
         controller: 'OrgEditModalCtrl',
-        scope: $scope
+        scope: scope
       });
 
-      $scope.$on('$stateChangeStart', function() {
+      states.onStateChangeStart($scope, true, function() {
         modal.dismiss('stateChange');
       });
 
       return modal;
     }
   };
-}).controller('OrgEditModalCtrl', function(api, forms, $modalInstance, orgs, $scope, $stateParams) {
+}).controller('OrgEditModalCtrl', function(api, forms, orgs, $scope, $uibModalInstance) {
 
-  $scope.organization = {};
+  $scope.organization = $scope.organization || {};
   $scope.editedOrg = {};
 
-  if ($stateParams.orgName) {
+  if ($scope.organization && $scope.organization.id) {
+    // Edit the specified organization.
+    reset();
```

Comment:
It is a bit confusing to have a reset function with a comment that says we edit the specified organization. It took me few minutes to understand that we set the form model to the correct organization depending the situation (we have an organization object, or an organization id meaning we have to retrieve the organization from server and then to set the form model).

I would suggest to rename the function `reset` to something like `setEditedOrganization`. We can keep `$scope.reset = <fnNewName>` as I suppose the button reset is bound to this `reset` fn.